gui/jails/forms.py

```diff
@@ -405,12 +416,28 @@ def save(self):
                     jail_flags |= WARDEN_SET_FLAGS_FLAGS
                     jail_set_args['jflags'] = val
 
-                jail_set_args['flags'] = jail_flags
-                try:
-                    w.set(**jail_set_args)
-                except Exception as e:
-                    self.errors['__all__'] = self.error_class([_(e.message)])
-                    return
+            else:
+                if key == 'jail_mac':
+                    duplicate_mac = 0
```

Comment:
This should be set to 0 inside the while loop else even if a single duplicate mac is generated then all succeeding ones will also be qualified as dups since this variable is not reset to 0 at the beginning of the while loop